prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -410,6 +411,65 @@ class ActionHistory {
         )
     }
 
+    // Used as temp storage by the json generator, below.
+    private data class DestinationData(
+        val orgSvc: OrganizationService,
+        var count: Int,
+        val sendingAt: OffsetDateTime? = null,
+    )
+
+    /**
+     * Generate nice json describing the destinations, suitable for returning to a Hub client.
+     * Most of the ugliness here is the attempt to not print every 1-entry report, but combine and summarize them.
+     *
+     * This works by side-effect on jsonGen.
+     */
+    fun prettyPrintDestinationsJson(jsonGen: JsonGenerator) {
```

Comment:
Currently the code generating the api/reports response JSON is in ReportFunction.   However, the new ActionHistory container class now has pretty much all the info needed to generate detailed responses.  A future refactor would be to move the response generation code fully into here.   This function is a first step in that direction.